### PR TITLE
refactor: refactor raft engine backend and state store

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

1. refactor RaftEngineBackend Error to common_meta::error::Error
2. refactor state store

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/log-store/src/raft_engine/backend.rs [hunk 1]
```diff
@@ -41,9 +42,9 @@ pub struct RaftEngineBackend {
 
 #[async_trait::async_trait]
 impl TxnService for RaftEngineBackend {
-    type Error = error::Error;
+    type Error = meta_error::Error;
```
Comment (MichaelScofield) [inline src/log-store/src/raft_engine/backend.rs:45]: Does the error type has to be changed to `meta_error::Error`? It introduces too much snafu context inside. A little ugly.

Comment (WenyXu) [reply src/log-store/src/raft_engine/backend.rs:45]: It's better to use `meta_error::Error`, To avoid having too many `KvBackendAdapter` s.

**Comment by WenyXu** [comment]
@MichaelScofield @fengjiachun PTAL

**Review by fengjiachun — APPROVED** [review]
LGTM

**Review by MichaelScofield — APPROVED** [review]

**Pull request merged**